lib/api/controllers/writeController.js

```diff
@@ -205,6 +205,33 @@ function WriteController (kuzzle) {
         return kuzzle.pluginsManager.trigger('data:afterCreateCollection', new ResponseObject(modifiedRequestObject, response));
       });
   };
+
+  /**
+   * Validate a document against collection validation specifications.
+   *
+   * @param {RequestObject} requestObject
+   * @returns {Promise} promise resolved as a ResponseObject
+   */
+  this.validateDocument = function validateDocument (requestObject) {
+    var modifiedRequestObject = null;
+
+    return kuzzle.pluginsManager.trigger('data:beforeValidateDocument', requestObject)
+      .then(newRequestObject => {
+        modifiedRequestObject = newRequestObject;
+        return kuzzle.validation.validationPromise(modifiedRequestObject, true);
+      })
+      .then(response => {
+        var responseObject = new ResponseObject(modifiedRequestObject, response);
+
+        if (!response.validation) {
+          responseObject.status = 400;
+          responseObject.error = response.errorMessages;
+        }
+        kuzzle.pluginsManager.trigger('log:error', `The document does not comply with the ${requestObject.index} / ${requestObject.collection} : ${JSON.stringify(requestObject.data.body)}`);
```

Comment:
I think that rejecting documents because they are not valid is not an error nor a warning, it's just normal activity to me.
If you insist on logging it, then I wouldn't use anything higher than `log:silly`, as this log can be very cost-heavy on large applications.